packages/core/src/compiler/Style.ts

```diff
@@ -1104,7 +1104,7 @@ const matchStyArgToSubArg = (
   varEnv: Env,
   styArg: PredArg<A> | SelExpr<A>,
   subArg: SubPredArg<A> | SubExpr<A>
-): Subst | undefined => {
+): Subst[] => {
```

Comment:
Could you clarify the reasoning behind this change? Is it just to make it easier to use this function later (avoiding an `if` statement)?

Reply:
Or does it sometimes return an array with more than one element?

Reply:
It does sometimes return an array with more than one matching! This is because of possible nested symmetric predicates. In particular, when matching a StyArg against a SubArg, the arg could be a predicate application, which might involve symmetry!